src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -463,19 +468,25 @@ export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
             let deselect = selected && oldTagOperation === TagOperationMode.None;
 
             // Only fire click event if a region is selected
-            if (this.props.selectedRegions &&
-                this.props.selectedRegions.length > 0 &&
-                this.props.onTagClick) {
-                deselect = false;
-                this.props.onTagClick(tag);
+            const { selectedRegions, onTagClick } = this.props;
+            if (selectedRegions && selectedRegions.length && onTagClick) {
```

Comment:
The only condition needed is: "selected region equals tag category or tag is free" right?

Could modify for simplicity and re-usability. If we add more non compatible region categories, 
`!this.isSelectionMark(selectedRegions[0].category) && !this.isSelectionMark(tag.type)`
will become a problem